### PR TITLE
Java 21 and Springboot 3 migration with terraform

### PR DESCRIPTION
**Pull request by dgurjar-ch**

### CODE REVIEW

File: document-generator-accounts/pom.xml
Comment (dhawes-ch) [inline document-generator-accounts/pom.xml:1]: Utilises structured-logging but there's no spring-boot-starter-logging dependency - https://companieshouse.atlassian.net/wiki/spaces/JU/pages/4901634070/StructuredLogger+format+missing+context

File: pom.xml
Comment (dhawes-ch) [inline pom.xml:1]: https://companieshouse.atlassian.net/wiki/spaces/JU/pages/4901634070/StructuredLogger+format+missing+context

File: pom.xml [hunk 1]
```diff
@@ -11,26 +11,22 @@
...
-        <maven-compiler-plugin.version>3.5.1</maven-compiler-plugin.version>
-        <maven-surefire-plugin.version>2.21.0</maven-surefire-plugin.version>
-        <maven-jar-plugin.version>2.4</maven-jar-plugin.version>
-        <junit-platform-surefire-provider.version>1.2.0</junit-platform-surefire-provider.version>
+        <maven-compiler-plugin.version>3.13.0</maven-compiler-plugin.version>
+        <maven-surefire-plugin.version>3.3.1</maven-surefire-plugin.version>
+        <maven-jar-plugin.version>3.4.1</maven-jar-plugin.version>
+        <junit-platform-surefire-provider.version>1.3.2</junit-platform-surefire-provider.version>
 
         <!-- Repositories -->
```
Comment (dhawes-ch) [inline pom.xml:31]: Think these are defined in the parent pom now

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java [hunk 2]
```diff
@@ -76,8 +76,7 @@ public Transaction getTransaction(String transactionLink, String requestId) thro
      * @return a statically created rest template
      */
     private static RestTemplate createRestTemplate() {
-        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
-        return new RestTemplate(requestFactory);
+        return new RestTemplate();
```
Comment (dhawes-ch) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java:79]: Why are we opting for this default constructor over the [one that takes the factory?](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#%3Cinit%3E(org.springframework.http.client.ClientHttpRequestFactory)) Does this not change the way the client behaves in some way?

Comment (dgurjar-ch) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java:79]: No It does not , We have discussed on call as well with you and Abhi, There was error  if request factory is used in constructor . So have to used default .

File: pom.xml (hunk 1 above)
Comment (dgurjar-ch) [reply pom.xml:31]: Yes , its parent POM.

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java (hunk 2 above)
Comment (dhawes-ch) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java:79]: I wonder if the error was due to an incorrect HttpComponents version being used as noted [on these docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.html)

Additionally Spring docs actually suggest [RestClient](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestClient.html) now which might be a better option

**Review by dhawes-ch — APPROVED** [review]

**Review by gjackson-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**